### PR TITLE
Add a intent parser that parses a text into an intent  and then execu…

### PR DESCRIPTION
**Pull request by tamarahills**
…tes the command.

### CODE REVIEW

**Comment by andrenatal** [comment]
@tamarahills you need to run `yarn instal` and commit `yarn.lock` in order to make travis pass

File: src/controllers/commands_controller.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+/**
+ * Parses the intent for a text sentence and sends to the API.ai intent
+ * parser to determine intent.  Then executes the intent as an action on the
+ * thing API.
+ */
+CommandsController.post('/', function (request, response) {
+
+  if(!request.body || request.body['text'] === undefined) {
+    response.status(400).send('Text element not defined');
```
Comment (hobinjk) [inline src/controllers/commands_controller.js]: As an API endpoint, errors should be sent back in a consistent JSON format like `{error: 'Message goes here'}`.

File: src/controllers/commands_controller.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+};
+
+/**
+ * Parses the intent for a text sentence and sends to the API.ai intent
+ * parser to determine intent.  Then executes the intent as an action on the
+ * thing API.
+ */
+CommandsController.post('/', function (request, response) {
+
+  if(!request.body || request.body['text'] === undefined) {
```
Comment (hobinjk) [inline src/controllers/commands_controller.js]: `request.body['text']` should be written as `request.body.text`

File: src/controllers/commands_controller.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+/**
+ * Formats the body for the request to API.ai and returns the String
+ */
+function getAiBody(text) {
+  let body = {
+    'v':'20150910',
+    'query': '',
+    'lang':'en',
+    'sessionId':'b083ae95-e39a-4954-a05b-84687f50f790',
+    'timezone':'2017-07-24T10:45:52-0400'
```
Comment (hobinjk) [inline src/controllers/commands_controller.js]: Is hardcoding this okay?

File: src/router.js [hunk 4]
```diff
@@ -76,6 +76,8 @@ var Router = {
       require('./controllers/users_controller'));
     app.use(API_PREFIX + Constants.UPLOADS_PATH,
       auth, require('./controllers/uploads_controller'));
+    app.use(API_PREFIX + Constants.COMMANDS_PATH,
```
Comment (hobinjk) [inline src/router.js]: missing `auth` middleware

File: src/controllers/commands_controller.js (hunk 2 above)
Comment (tamarahills) [reply src/controllers/commands_controller.js]: fixed.

File: src/controllers/commands_controller.js (hunk 1 above)
Comment (tamarahills) [reply src/controllers/commands_controller.js]: Added these as a message in JSON format.

File: src/router.js (hunk 4 above)
Comment (tamarahills) [reply src/router.js]: fixed.

File: src/controllers/commands_controller.js (hunk 3 above)
Comment (tamarahills) [reply src/controllers/commands_controller.js]: I removed this as it's optional and we don't have anything depending on TZ ATM.

**Pull request merged**